### DISCUSSION TITLE
How to pass a opentracing.Span with the context through a gRPC call with go-micro

### DISCUSSION THREAD

**Discussion by djedjethai** [micro/go-micro#2548] [Q&A] [answered]
Hi,

How to pass an opentracing.Span id between two services(2 handlers) using a gRPC call, see:

```
// before sending
ct := opentracing.ContextWithSpan(ctx, sp)
fmt.Println(ct)
// you see the span-id is inside the context
.WithValue(type opentracing.contextKey, val 0de0a6cf7b8c66d1:0de0a6cf7b8c66d1:0000000000000000:1)
```
   
But then on the receiver handler
```
func (s *handler) Get(ctx context.Context, req *pb.GetRequest, res *pb.GetResponse) error {
sp = opentracing.SpanFromContext(ctx)
fmt.Println("with span: ", sp)
}
with span:  {}
// and even I println the ctx, the span-id is gone
```
The span identifier is gone... ?? How can I pass it ?

**Comment by Davincible** [comment, marked as answer]
Contexts are Go instances and cannot simply be send over a grpc call. You'd have to send the span separately from the context

**Comment by djedjethai** [comment]
Thanks for the Information @Davincible, I finally tried to add them to the body and it works.
